CIP-????/README.md

```diff
@@ -0,0 +1,132 @@
+
+- [Cardano-Reference-Scripts](https://github.com/fallen-icarus/cardano-reference-scripts) - trustless p2p sharing of Cardano reference scripts. It uses the UTxO info API to broadcast all of the UTxOs that contain each beacon. The beacon token name is the script hash of the reference script being shared. Minting the beacon requires that the beacon is stored with the reference script whose hash matches the beacon's token name. This example also uses a helper plutus spending script to guarantee the burning of the beacon when a reference script UTxO is consumed. 
+
+### Distributed L1 Dapps
+As already alluded to, a *distributed L1 Dapp* is one where all users have their own personal address while using the Dapp. This is opposite to *concentrated L1 Dapps* where all users share one or a few addresses while using the Dapp.
+
+Distributed L1 Dapps can use beacon tokens to aggregrate all of the necessary information for using the Dapps. Such distributed L1 Dapp follow the same general design pattern:
+
+1. All user addresses use the same spending script for a given use case.
+2. All user addresses *must* have a staking credential.
+3. The spending script delegates the authorization of owner related actions to the address' staking credential - ie, the staking key must sign or the staking script must be successfully executed in the same transaction.
+4. The spending script's hash is hard-coded into the beacon minting policy and the minting policy enforces that the beacons can only be minted to an address of that spending script.
+5. The datums for the spending script must have the policy ID of the beacons so that the script can force proper usage of the beacons (such as burning when necessary).
+
+#### Advantages
+Since each user has his/her own address for the Dapp, the following are now possible:
+
+1. **Full delegation control**
+2. Users maintain full custody of their assets while using the Dapp, since owner related actions must be approved by the staking credential.
+2. The address itself can act as the User ID. There is no need to place a User ID in a datum and guard its usage.
+
+Furthermore, the Dapp itself gains some nice features:
+
+1. Since there are at least as many UTxOs as there are users, the distributed L1 Dapp is naturally concurrent and gets more concurrent as the number of users increases.
+2. Upgradability can happen democratically - ie, users can choose whether to move their assets to an address using a newer version of the Dapp's spending script.
+3. The Dapp is easily integratable into existing frontends (wallets).
+4. Since the address itself can act as the User ID, in some cases, the Dapp's logic can be dramatically simplified.
+
+#### Cardano-Swaps: A L1 DEX with full delegation control
```

Comment:
Considering that this may be the most commercially interesting & community supported reference implementations, can you combine it with the 2 above for a list of 3 total?  Do they need to be inventoried separately?  If not, I think it would be more compelling & readable if combined in a single section of reference implementations & exactly how Beacon Tokens are used in each of them.